suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,282 @@
-#     ...
-#
-#
-# handlers = []
+from collections import defaultdict
+from pathlib import Path
+from tifffile import TiffWriter
+import event_model
+import numpy
+import suitcase.utils
+from ._version import get_versions
+
+__version__ = get_versions()['version']
+del get_versions
+
+
+def export(gen, directory, file_prefix='{uid}-', stack_images=True, **kwargs):
+    """
+    Export a stream of documents to TIFF stack(s).
+
+    This creates a file named:
+    ``<directory>/<file_prefix>{stream_name}-{field}.tiff``
+    for every Event stream and field that contains 2D 'image like' data.
+
+    .. warning::
+
+        This process explicitly ignores all data that is not 2D and does not
+        include any metadata in the output file.
+
+    .. note::
+
+        This can alternatively be used to write data to generic buffers rather
+        than creating files on disk. See the documentation for the
+        ``directory`` parameter below.
+
+    Parameters
+    ----------
+    gen : generator
+        expected to yield ``(name, document)`` pairs
+
+    directory : string, Path or Manager.
+        For basic uses, this should be the path to the output directory given
+        as a string or Path object. Use an empty string ``''`` to place files
+        in the current working directory.
+
+        In advanced applications, this may direct the serialized output to a
+        memory buffer, network socket, or other writable buffer. It should be
+        an instance of ``suitcase.utils.MemoryBufferManager`` and
+        ``suitcase.utils.MultiFileManager`` or any object implementing that
+        inferface. See the suitcase documentation (LINK ONCE WRITTEN) for
+        details.
+
+    file_prefix : str, optional
+        The first part of the filename of the generated output files. This
+        string may include templates as in ``{proposal_id}-{sample_name}-``,
+        which are populated from the RunStart document. The default value is
+        ``{uid}-`` which is guaranteed to be present and unique. A more
+        descriptive value depends on the application and is therefore left to
+        the user.
+
+    stack_images : Boolean
+        This indicates if we want one image per file (`stack_images` = `False`)
+        or many images per file (`stack_images` = `True`). If using
+        `stack_images` = `False` then an additional image number is added to
+        the file name.
+
+    **kwargs : kwargs
+        kwargs to be passed to ``tifffile.TiffWriter.save``.
+
+    Returns
+    -------
+    dest : dict
+        dict mapping the 'labels' to lists of file names
+
+    Examples
+    --------
+
+    Generate files with unique-identifer names in the current directory.
+
+    >>> export(gen, '')
+
+    Generate files with more readable metadata in the file names.
+
+    >>> export(gen, '', '{plan_name}-{motors}-')
+
+    Include the experiment's start time formatted as YY-MM-DD_HH-MM.
+
+    >>> export(gen, '', '{time:%%Y-%%m-%%d_%%H:%%M}')
```

Comment:
This example, like the others, should have a trailing `-` after the formatted string to separate the timestamp from the stream name. (This is my mistake, of course!)